src/transport/SecureMessageCodec.cpp

```diff
@@ -43,22 +43,18 @@ CHIP_ERROR Encrypt(const CryptoContext & context, CryptoContext::ConstNonceView
     VerifyOrReturnError(!msgBuf->HasChainedBuffer(), CHIP_ERROR_INVALID_MESSAGE_LENGTH);
     VerifyOrReturnError(msgBuf->TotalLength() <= kMaxAppMessageLen, CHIP_ERROR_MESSAGE_TOO_LONG);
 
-    static_assert(std::is_same<decltype(msgBuf->TotalLength()), uint16_t>::value,
-                  "Addition to generate payloadLength might overflow");
-
     ReturnErrorOnFailure(payloadHeader.EncodeBeforeData(msgBuf));
 
-    uint8_t * data    = msgBuf->Start();
-    uint16_t totalLen = msgBuf->TotalLength();
+    uint8_t * data  = msgBuf->Start();
+    size_t totalLen = msgBuf->TotalLength();
 
     MessageAuthenticationCode mac;
     ReturnErrorOnFailure(context.Encrypt(data, totalLen, data, nonce, packetHeader, mac));
 
     uint16_t taglen = 0;
     ReturnErrorOnFailure(mac.Encode(packetHeader, &data[totalLen], msgBuf->AvailableDataLength(), &taglen));
 
-    VerifyOrReturnError(CanCastTo<uint16_t>(totalLen + taglen), CHIP_ERROR_INTERNAL);
-    msgBuf->SetDataLength(static_cast<uint16_t>(totalLen + taglen));
+    msgBuf->SetDataLength(static_cast<size_t>(totalLen + taglen));
```

Comment:
Why does this need a cast?  I would expect the type of `totalLen + taglen` to already be `size_t`, since it's the size of a `size_t` and `uint16_t`, both are unsigned, and `size_t` is at least 16 bits.

Reply:
Yes, this cast is redundant. Have removed it.